src/Models/NonhydrostaticModels/pressure_correction.jl

```diff
@@ -10,11 +10,11 @@ function calculate_pressure_correction!(model::NonhydrostaticModel, Δt)
     # Mask immersed velocities
     foreach(mask_immersed_field!, model.velocities)
 
-    fill_halo_regions!(model.velocities, model.clock, fields(model))
+    fill_halo_regions!(model.velocities, model.clock, fields(model); Δt)
```

Comment:
Why use a keyword argument?

Another possibility is to add `Δt` to `clock` (since this only requires generalizing the definition of `clock`, as well as `tick!`)

Reply:
So I recommend either adding `Δt` to clock, or using

```julia
fill_halo_regions!(model.velocities, model.clock, Δt, fields(model))
```

the latter case is a major breaking change, but I think if we need the time-step for `Open` then its plausible that users will need it for other reasons.